sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -252,6 +252,8 @@ statement
         '(' columns=multipartIdentifierPropertyList ')'
         (OPTIONS options=propertyList)?                                #createIndex
     | DROP INDEX (IF EXISTS)? identifier ON TABLE? multipartIdentifier #dropIndex
+    | CREATE TEMPORARY MACRO IDENTIFIER '(' colTypeList ')'
```

Comment:
should be:
"| CREATE TEMPORARY MACRO IDENTIFIER '(' colTypeList? ')'"
to support no arg macro.
The colTypeList is optional.